### PR TITLE
feat: allow hero to be styled with palette

### PR DESCRIPTION
**Pull request by meghaniankov**
# Description

Allow Hero component background color to be set with a Palette in Contentful

image

image

# Checklist

Pull request contains:

- [ ] A new component
- [x] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

Definition of done:

- [ ] Includes theme changes for both Uswitch and money
- [ ] Work has been tested in multiple browsers
- [x] PR description includes description of change
- [x] PR description includes screenshot of change
- [ ] If new component, designer has approved screenshot
- [ ] If the change will affect other teams, that team knows about this change
- [x] If introducing a new component behaviour, added a story to cover that case.

### CODE REVIEW

File: src/elements/hero/src/index.tsx [hunk 1]
```diff
@@ -55,92 +56,97 @@ const Hero: React.FC<Props> = ({
...
-          position: 'relative',
-          overflow: 'hidden',
-          variant: 'elements.hero.wrapper',
-          backgroundColor: customBgColor || undefined
-        }}
-        className={className}
+      <Palette
+        as="div"
+        sx={{ variant: 'elements.hero.wrapper', overflow: 'hidden' }}
+        px={{ backgroundColor: 'backgroundColor' }}
```
Comment (Segger87) [inline src/elements/hero/src/index.tsx:62]: your setting backgroundColor to a string called backgroundColor is that intentional?

File: src/elements/hero/src/index.tsx [hunk 2]
```diff
@@ -55,92 +56,97 @@ const Hero: React.FC<Props> = ({
...
+                  right: 0,
+                  top: 0,
+                  bottom: 0,
+                  display: [
+                    fgImageOnMobile ? 'block' : 'none',
+                    fgImageOnTablet ? 'block' : 'none',
+                    'block'
+                  ],
+                  ...(fgImage && fgImageType === 'background'
+                    ? {
```
Comment (Segger87) [inline src/elements/hero/src/index.tsx:100]: should the spread be in front of the object on line 100?

File: src/elements/hero/src/index.tsx (hunk 1 above)
Comment (meghaniankov) [reply src/elements/hero/src/index.tsx:62]: Yep, it's tied to the Palette 'backgroundColor' field in Contentful

File: src/elements/hero/src/index.tsx (hunk 2 above)
Comment (meghaniankov) [reply src/elements/hero/src/index.tsx:100]: I didn't add this, this was from Callum. It doesnt work when I move the spread operator to line 100

**Review by Segger87 — DISMISSED** [review]

**Review by jonathan-fielding — DISMISSED** [review]

**Review by jonathan-fielding — APPROVED** [review]

**Pull request merged**